### ISSUE TITLE
Skipped files are shown as warnings

### ISSUE THREAD

**Issue by ReneRebsdorf** [github/super-linter#2391]
## Describe the bug

If a file is not processed, it is shown as a warning:

``` text
... ISO27002/A.14 - System Acquisition, Development & Maintenance/14.2.8 System security testing.md} existed in commit data, but not found on file system, skipping ...
```

## Expected behavior

As markdown file was not processed, which was due to special characters in the folder name ('&'). I would expect this to be 'assumed-bad' and fail in linting.
Maybe there's a flag for this, that I am not aware of, treating warnings as errors maybe?

## Steps to Reproduce

- Create a folder with the name 'abc & def' and containing a markdown file
- Commit the file locally
- Run the tests using the below
- Observe the output only contains WARN, where an error was expected.

``` text
docker run \
  -e RUN_LOCAL=true \
  -e VALIDATE_MARKDOWN=true \
  -v $(pwd):/tmp/lint \
  ghcr.io/github/super-linter:v3.13.5
```

**Comment by admiralAwkbar** [comment]
@ReneRebsdorf 
Hey! Can you try to run this on the 4.x release and see if you get better behavior?

example:
```bash
docker run \
  -e RUN_LOCAL=true \
  -e VALIDATE_MARKDOWN=true \
  -v $(pwd):/tmp/lint \
  ghcr.io/github/super-linter:v4
```
I know we have made some improvements but if still an issue I can easily troubleshoot and find a fix :)

**Comment by ReneRebsdorf** [comment]
@admiralAwkbar Sorry, my mistake.  We were pulling the latest image at the time of writing, not 3.13.5 as I put in the bug.
This was tested with image: b114f238e839, e.g. v.4.8.5

**Comment by ReneRebsdorf** [comment]
Reproducible with v.4.8.6 as well.

**Comment by admiralAwkbar** [comment]
@ReneRebsdorf Thank you for checking, i will look into the source and see what I can find :)

**Comment by admiralAwkbar** [comment]
@ReneRebsdorf 

I tried to reproduce the error but I think i know whats happening...

I used the latest version of the `GitHub/Super-Linter:v4`

![Screen Shot 2022-01-27 at 3 00 03 PM](https://user-images.githubusercontent.com/29484535/151442798-706583ea-c8a7-4509-8ef0-d435f9833b9f.png)

As you can see it was able to find the file: `abc & 123/file.md`
It should work as we **double quote** all our pathing.

The specific code block  were hitting:
```bash
##########################################################
    # Check if the file exists on the filesystem, or skip it #
    ##########################################################
    if [ ! -f "${FILE}" ]; then
      # File not found in workspace
      warn "File:{$FILE} existed in commit data, but not found on file system, skipping..."
      continue
    fi
```
This is a check to make sure we don't fail to look for a file that showed up in the history of your report but may have been moved or deleted.

Can you validate that this is the case?

**Comment by ReneRebsdorf** [comment]
@admiralAwkbar Thank you for looking into this so promptly.

I cannot validate this is the case however - maybe I am misunderstanding your question?

The folder which was named 'A.14 - System Acquisition, Development & Maintenance' was existing at the time of execution
Same was the sample I made to determine it was the '&' that was causing issues, so I did a local commit a the test file within the 'abc & def' folder I made, and was immediately able to reproduce, without moving any files during test runs.

The odd thing is when you look at the names, as mentioned the folder name is `A.14 - System Acquisition, Development & Maintenance`, but according to the log, it is recognized as: `/tmp/lint/docs/30-ISO27002/A.14 - System Acquisition, Development`

In other words, everything after the '&' is not evaluated.

**Comment by admiralAwkbar** [comment]
@ReneRebsdorf Very interesting
i will create the same folder name and test more :)
Seems I may have not tried hard enough :)

**Comment by admiralAwkbar** [comment]
![Screen Shot 2022-01-28 at 8 32 24 AM](https://user-images.githubusercontent.com/29484535/151564845-e6119895-bde8-4059-b26b-b940d24655f4.png)
Man... @ReneRebsdorf It won't let me reproduce this error for you lol...

**Comment by ReneRebsdorf** [comment]
@admiralAwkbar 

That is super strange.
For my own sanity sake I created new folders, in different locations, output was always:

![image](https://user-images.githubusercontent.com/44376705/151580821-1b1cc157-88e0-49f5-8dcc-00d307435140.png)

Note, the folder name in all cases is 'abc & def' (not just 'abc').

I can reproduce on both our CI (Azure DevOps, running the 'ubuntu-latest' image), and locally (Win11).
Not that it should matter that much for docker reasons.

**Comment by admiralAwkbar** [comment]
I'm starting to wonder if this is a windows related issue...
Are you running this on Windows?

**Comment by ReneRebsdorf** [comment]
I've tried commiting and running files from both Linux and Windows to verify if OS indeed has something to do with it.

I am able to reproduce this on:

- Ubuntu 20.04, ran from a Microsoft hosted Azure DevOps build agent
- Kali running from with WSL
- Windows 11 
- Windows 10

Unfortunately I wasn't able to fix the problem with by using a different OS.

**Comment by admiralAwkbar** [comment]
@ReneRebsdorf I'm sorry I wasn't able to reproduce this... I'll think about it some more and see if I can figure out the corner case. Again very sorry for the inconvenience...

**Comment by ReneRebsdorf** [comment]
@admiralAwkbar No worries! We have since changed from the ampersand to just using the word 'and', so this is not of urgency.
Similar problems in future could however cause the linter to provide false passes, which this issue was intended to help bring focus towards, along with a hopeful code change.

I'll try to find time to further determine how we are experiencing very different behavior.
